charts/seed-operatingsystemconfig/original/templates/kubelet/_kubelet-config-file

```diff
@@ -19,7 +19,11 @@ cgroupsPerQOS: true
 cgroupDriver: cgroupfs
 clusterDomain: {{ required "kubernetes.domain is required" .Values.kubernetes.domain }}
 clusterDNS:
+{{- if and .Values.kubernetes.ipvsEnabled .Values.kubernetes.nodeLocalDNSEnabled}}
+- {{ required "kubernetes.nodeLocalDNS is required" .Values.kubernetes.nodeLocalDNS }}
+{{- else }}
 - {{ required "kubernetes.clusterDNS is required" .Values.kubernetes.clusterDNS }}
+{{- end }}
```

Comment:
The `ipvsEnabled `, `nodeLocalDNSEnabled`, and `nodeLocalDNS` variables are not needed. You can simply pass the node-local DNS cache IP as part of the `clusterDNS` variable and get the same result.

Reply:
Not sure if the order is preserved, according to the documentation of the [cluster-dns](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet)  flag:
 
> There is no guarantee as to which DNS server may be contacted for name resolution

Reply:
Hm, which order? In both cases if the `if-else` block you only add exactly one IP. My point was that it doesn't help to have multiple variables here. Just keep `clusterDNS` and set it accordingly when rendering the Helm chart.

Reply:
then the logic of which value will be computed in the code with a condition right?

Reply:
Yes, probably. The benefit would be that you can more easily write unit tests for the function that computes the chart values. OTOH, I agree that this is still not optimal because even if the Helm chart values are correctly computed/tested the Helm chart itself must still be correct as well, and this is not that straight-forward to test. :(
Anyway, for this particular code snippet I think it's more clear to have only one variable that is filled with the correct value by the caller.